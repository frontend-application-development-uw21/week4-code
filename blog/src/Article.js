import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Article extends Component {
    static propTypes = {
        article: PropTypes.shape({
            author: PropTypes.shape({
                name: PropTypes.string.isRequired,
                image: PropTypes.string.isRequired,
                isMediumMember: PropTypes.bool.isRequired
            }).isRequired,
            description: PropTypes.string.isRequired,
            hasAudioAvailable: PropTypes.bool.isRequired,
            image: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired,
            // Whether an article is behind a paywall
            memberPreview: PropTypes.bool.isRequired,
            minutesToRead: PropTypes.number.isRequired,
            postedDate: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired
        }).isRequired
    }

    render() {
        const { title, description, hasAudioAvailable } = this.props.article;
        // let audioAvailable;
        // if (hasAudioAvailable) {
        //     audioAvailable = (
        //         <div>
        //             Audio Available
        //         </div>
        //     )
        // }

        return (
            <div>
                <h2>{title}</h2>
                <p>{description}</p>
                {hasAudioAvailable && (
                    <div>
                        Audio Available
                    </div>
                )}
            </div>
        )
    }
}

export default Article;