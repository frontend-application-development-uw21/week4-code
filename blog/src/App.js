import './App.css';
import Article from './Article';
import missedArticles from './_data/missed-articles.json';

function App() {
  return (
    <div>
      <h1>Blog</h1>
      {/* <Article /> */}
      {missedArticles.map((a, idx) => {
        return <Article key={idx} article={a} />
      })}
      <h1>My Blog</h1>
      <Article article={{
        title: 'You wont believe this story',
        description: 'Click here to read more',
        hasAudioAvailable: false
      }} />
    </div>
  );
}

export default App;
