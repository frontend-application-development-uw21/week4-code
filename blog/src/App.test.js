import App from './App';
import Article from './Article';
import { shallow, render } from 'enzyme';

test('rendering the entire app', () => {
  const component = render(<App />);

  expect(component).toMatchSnapshot();
});

it('should render 3 articles', () => {
  const component = shallow(<App />);
  const articles = component.find('Article');
  expect(articles).toHaveLength(3);
});

function circleArea(radius) {
  return 3.14 * radius * radius;
}

it('should return correct area', () => {
  expect(circleArea(4)).toEqual(50.24);
})