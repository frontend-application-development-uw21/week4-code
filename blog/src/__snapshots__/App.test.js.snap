// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering the entire app 1`] = `
<div>
  <h1>
    Blog
  </h1>
  <div>
    <h2>
      What Would Happen If You Did These Things Every Day...
    </h2>
    <p>
      For thirty days...
    </p>
  </div>
  <div>
    <h2>
      10 Strategies for Creating Viral and Powerful Content
    </h2>
    <p />
  </div>
  <div>
    <h2>
      Useful Tools for Writers
    </h2>
    <p>
      Jumpstart your process.
    </p>
  </div>
  <h1>
    My Blog
  </h1>
</div>
`;
